### PR TITLE
Serve libraries locally in the docker container

### PR DESCRIPTION
**Pull request by kaysond**
Addresses #93 

Created a `static` directory and the `index_local.html` that gets copied in the container. I put all of the libraries in `static/libraries.txt` and the container automatically downloads all of the files in that list.

I had to manually add/modify the google fonts css file since it includes font imports that also need to be downloaded.

I also cleaned up the tcp_server a little in terms of routing.

Everything works fine, but I am getting some warnings on build:
```
Step 13/26 : RUN cargo build --release -p lldap     && ./app/build.sh 
 ---> Running in ae86bb0e19da
   Compiling lldap_auth v0.2.0 (/app/auth)
   Compiling lldap v0.2.0 (/app/server)
    Finished release [optimized] target(s) in 2m 20s
[INFO]: Checking for the Wasm target...
[INFO]: Compiling to Wasm...
   Compiling lldap_auth v0.2.0 (/app/auth)
   Compiling lldap_app v0.2.0 (/app/app)
    Finished release [optimized] target(s) in 57.40s
[WARN]: :-) origin crate has no README
[INFO]: Installing wasm-bindgen...
[INFO]: Optimizing wasm binaries with `wasm-opt`...
[INFO]: Optional fields missing from Cargo.toml: 'description', 'repository', and 'license'. These are not necessary, but recommended                                                       [INFO]: :-) Done in 2m 07s
[INFO]: :-) Your wasm pkg is ready to publish at /app/app/pkg.

./main.js → ./pkg/bundle.js...
(!) Unresolved dependencies
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
bootstrap (imported by pkg/lldap_app.js)
(!) Missing global variable name
Use output.globals to specify browser global variable names corresponding to external modules
bootstrap (guessing 'bootstrap')
created ./pkg/bundle.js in 708ms
```
Not sure if this is old or caused by making bootstrap local.

### CODE REVIEW

**Comment by kaysond** [comment]
A couple of things occurred to me with this approach:
- source maps don't work because we're not using the cdn
- it in theory introduces an attack vector that didn't exist before because we're not checking hashes (I had to remove them because for some reason they didn't match...)

**Comment by kaysond** [comment]
Also found another issue - the bootstrap icons font is expected to be at `/static/fonts/...` but the way the dockerfile is set up, it ends up at `/static`.

**Comment by nitnelave** [comment]
Do you need help with that or do you want to keep investigating on your side?

**Comment by kaysond** [comment]
I'll keep working on it, just wanted to make sure nothing is way outside your expectation.

I do need your input on the rust error, though. I'm guessing `[WARN]: :-) origin crate has no README` is ok?

**Comment by kaysond** [comment]
alright everything is fixed and tested. this should be good for merging

File: server/src/infra/tcp_server.rs [hunk 1]
```diff
@@ -76,8 +70,16 @@ fn http_config<Backend>(
...
     .service(Files::new("/pkg", "./app/pkg"))
-    // Default to serve index.html for unknown routes, to support routing.
-    .service(web::scope("/").route("/.*", web::get().to(index)));
+    // Serve static files
+    .service(Files::new("/static", "./app/static"))
+    // Serve static fonts
+    .service(Files::new("/static/fonts", "./app/static/fonts"))
+    // Serve the index
+    .service(
+        web::scope("/")
```
Comment (nitnelave) [inline server/src/infra/tcp_server.rs:79]: We need to serve every 404 with the index, for routing purposes.
If the user loads the page "http://server.tld/user/bob" we need to serve the index, and the SPA logic will route to the correct page.

File: app/build.sh [hunk 2]
```diff
@@ -24,4 +24,4 @@ then
   exit 1
 fi
 
-$ROLLUP_BIN ./main.js --format iife --file ./pkg/bundle.js
+$ROLLUP_BIN ./main.js --format iife --file ./pkg/bundle.js --globals bootstrap:bootstrap
```
Comment (nitnelave) [inline app/build.sh:27]: Thanks for silencing the warning!

**Review by nitnelave — CHANGES_REQUESTED** [review]

File: server/src/infra/tcp_server.rs (hunk 1 above)
Comment (kaysond) [reply server/src/infra/tcp_server.rs:79]: I was wondering why that wildcard route was there. I was also wondering why every incorrect `/...` request failed. 🤦🏽‍♂️

Comment (nitnelave) [reply server/src/infra/tcp_server.rs:79]: Sorry, last nit: can you add back the comment?
// Default to serve index.html for unknown routes, to support routing.

**Review by nitnelave — APPROVED** [review]

File: server/src/infra/tcp_server.rs (hunk 1 above)
Comment (kaysond) [reply server/src/infra/tcp_server.rs:79]: Done!

**Review by nitnelave — APPROVED** [review]

**Comment by nitnelave** [comment]
Thanks a lot! Merged.

**Pull request merged**